### PR TITLE
[upgrade] Upgrading various metric monitoring components

### PR DESCRIPTION
**Pull request by cumcke**

### CODE REVIEW

File: monitoring/values-prom-operator.yaml [hunk 1]
```diff
@@ -87,7 +87,7 @@ prometheus:
     nodePort: null  
   prometheusSpec:
     image:
-      tag: v2.44.0
+    #  tag: v2.44.0
```
Comment (cumcke) [inline monitoring/values-prom-operator.yaml:90]: For whatever reason, the Helm chart did not go to the latest like I expected.  The Helm chart updates Prometheus to 2.45.0; however, when I did the open source components review, Prometheus was at 2.46.0.
Unless there are any objections, I will allow the Helm chart to update it (2.45.0) to reduce the number of manual steps to this task.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,24 @@
 # SAS Viya Monitoring for Kubernetes
 
+## Unreleased
+
+* **Overall**
+
+* **Metrics**
+  * [UPGRADE] - Kube-prometheus-stack has been upgraded from version 45.28.0 to 48.2.3
```
Comment (gsmith-sas) [inline CHANGELOG.md:8]: Typo: the new version is (I believe) 48.3.2.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,24 @@
...
+  * [UPGRADE] - Prometheus has been upgraded from version 2.44.0 to 2.45.0
```
Comment (gsmith-sas) [inline CHANGELOG.md:9]: In my test environment, after fresh install, Prometheus was (still) at 2.44.0.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,24 @@
...
+  * [UPGRADE] - Prometheus Operator has been upgraded from version 0.65.1 to 0.66.0
+  * [UPGRADE] - Grafana has been upgraded from version 9.5.5 to 10.0.3
+  * [UPGRADE] - Kube State Metrics has been upgraded from version 2.8.2 to 2.9.2
+  * [UPGRADE] - K8s-sidecar used with Grafana has been upgraded from version 1.5.0 to 1.6.0
```
Comment (gsmith-sas) [inline CHANGELOG.md:13]: In my test environment, after fresh install, k8s-sidecar containers were using image: quay.io/kiwigrid/k8s-sidecar:1.24.6

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,24 @@
...
+  * [UPGRADE] - Node Exporter has been upgraded from version 1.24.0 to 1.24.6
```
Comment (gsmith-sas) [inline CHANGELOG.md:14]: In my test environment, after fresh install, node-exporter containers were using image: quay.io/prometheus/node-exporter:v1.6.0

**Review by gsmith-sas — CHANGES_REQUESTED** [review]
Looks like this needs to be revised based on the changes we pushed as part of supporting Helm charts in tar format.   Specifically, with that PR, we stopped setting Helm chart versions in the code and moved setting those environment variables in a separate file: component_versions.env.

File: CHANGELOG.md (hunk 3 above)
Comment (gsmith-sas) [reply CHANGELOG.md:9]: Most likely an artifact of not picking up the component_versions.env file.  I suggest taking care of that first and see whether this goes away.

File: CHANGELOG.md (hunk 4 above)
Comment (gsmith-sas) [reply CHANGELOG.md:13]: Most likely an artifact of not picking up the component_versions.env file.  I suggest taking care of that first and see whether this goes away.

File: CHANGELOG.md (hunk 5 above)
Comment (gsmith-sas) [reply CHANGELOG.md:14]: Most likely an artifact of not picking up the component_versions.env file.  I suggest taking care of that first and see whether this goes away.

File: CHANGELOG.md (hunk 2 above)
Comment (cumcke) [reply CHANGELOG.md:12]: Yep, that was a typo.  It'll be fixed in the next push.

File: CHANGELOG.md (hunk 3 above)
Comment (cumcke) [reply CHANGELOG.md:13]: No, for some reason, the Helm chart is not picking up the correct version of the Prometheus in the kube-prometheus-stack [values.yaml](https://github.com/prometheus-community/helm-charts/blob/kube-prometheus-stack-48.3.2/charts/kube-prometheus-stack/values.yaml#L2897) file.  I'm going to manually update it to 2.46.0 for the time being.

File: CHANGELOG.md (hunk 4 above)
Comment (cumcke) [reply CHANGELOG.md]: I got versions for the K8s-sidecar and the Node Exporter mixed up.  They'll be corrected in the next push.

File: CHANGELOG.md (hunk 5 above)
Comment (cumcke) [reply CHANGELOG.md]: I got versions for the K8s-sidecar and the Node Exporter mixed up.  They'll be corrected in the next push.

File: CHANGELOG.md (hunk 3 above)
Comment (gsmith-sas) [reply CHANGELOG.md:13]: I am now seeing Prometheus 2.46.0 in my test environment.  Please update the CHANGELOG.md entry to reflect that version (if that is our intent).

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,12 +1,27 @@
...
 * **Overall**
   * [FEATURE] Deploying into an air-gapped environment using local Helm archive (.tgz) files is now supported.  In this scenario,
 container images are still expected to be pulled from private container registry.  See
 [Configure SAS Viya Monitoring for Kubernetes for an Air-Gapped Environment](https://documentation.sas.com/?cdcId=obsrvcdc&cdcVersion=default&docsetId=obsrvdply&docsetTarget=n0grd8g2pkfglin12bzm3g1oik2p.htm).
   * [FIX] Ensure forced updates of Helm repos actually occur
 
+* **Metrics**
+  * [UPGRADE] - Kube-prometheus-stack has been upgraded from version 45.28.0 to 48.3.2
+  * [UPGRADE] - Prometheus has been upgraded from version 2.44.0 to 2.45.0
+  * [UPGRADE] - Prometheus Operator has been upgraded from version 0.65.1 to 0.66.0
```
Comment (gsmith-sas) [inline CHANGELOG.md:14]: I am seeing Prometheus Operator version 0.67.1 and NOT 0.66.0.  Please update CHANGELOG.md or code.

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,12 +1,27 @@
...
+  * [UPGRADE] - Grafana has been upgraded from version 9.5.5 to 10.0.3
+  * [UPGRADE] - Kube State Metrics has been upgraded from version 2.8.2 to 2.9.2
+  * [UPGRADE] - K8s-sidecar used with Grafana has been upgraded from version 1.24.0 to 1.24.6
```
Comment (gsmith-sas) [inline CHANGELOG.md:17]: I am seeing quay.io/kiwigrid/k8s-sidecar:1.25.0 being deployed in my environment.  Please update CHANGELOG.md or code as appropriate.

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,12 +1,27 @@
...
+  * [UPGRADE] - Node Exporter has been upgraded from version 1.5.0 to 1.6.0
```
Comment (gsmith-sas) [inline CHANGELOG.md:18]: I am seeing quay.io/prometheus/node-exporter:v1.6.1 being deployed in my environment. Please update CHANGELOG.md or code as appropriate.

File: monitoring/openshift/grafana-values.yaml [hunk 9]
```diff
@@ -10,9 +10,9 @@ sidecar:
     enabled: true
     label: grafana_datasource
   image:
-    registry: quay.io
-    repository: kiwigrid/k8s-sidecar
-    tag: 1.24.0
+  #  registry: quay.io
+  #  repository: kiwigrid/k8s-sidecar
+    tag: 1.25.0
```
Comment (gsmith-sas) [inline monitoring/openshift/grafana-values.yaml:15]: Incorrect version number listed in CHANGELOG.md

File: monitoring/values-prom-operator.yaml [hunk 10]
```diff
@@ -18,9 +18,9 @@ commonLabels:
 # https://github.com/coreos/prometheus-operator
 prometheusOperator:
   image:
-    registry: quay.io
-    repository: prometheus-operator/prometheus-operator
-    tag: v0.65.1
+  #  registry: quay.io
+  #  repository: prometheus-operator/prometheus-operator
+    tag: v0.67.1
```
Comment (gsmith-sas) [inline monitoring/values-prom-operator.yaml:23]: Incorrect version number listed in CHANGELOG.md

File: monitoring/values-prom-operator.yaml [hunk 11]
```diff
@@ -87,7 +87,7 @@ prometheus:
     nodePort: null  
   prometheusSpec:
     image:
-      tag: v2.44.0
+      tag: v2.46.0
```
Comment (gsmith-sas) [inline monitoring/values-prom-operator.yaml:90]: Incorrect version number listed in CHANGELOG.md

File: monitoring/values-prom-operator.yaml [hunk 12]
```diff
@@ -151,9 +151,9 @@ alertmanager:
 # https://github.com/prometheus-community/helm-charts/tree/main/charts/prometheus-node-exporter
 prometheus-node-exporter:
   image:
-    registry: quay.io
-    repository: prometheus/node-exporter
-    tag: v1.5.0
+  #  registry: quay.io
+  #  repository: prometheus/node-exporter
+    tag: v1.6.1
```
Comment (gsmith-sas) [inline monitoring/values-prom-operator.yaml:156]: Incorrect version number listed in CHANGELOG.md

File: ARTIFACT_INVENTORY.md [hunk 13]
```diff
@@ -25,15 +25,15 @@ registry/repository/image_name:version
...
 | Metrics | Ingress-NGINX Controller | registry.k8s.io/ingress-nginx/kube-webhook-certgen:v20221220-controller-v1.5.1-58-g787ea74b6|
-| Metrics | Kube State Exporter | k8s.gcr.io/kube-state-metrics/kube-state-metrics:v2.8.2|
-| Metrics | Node Exporter | quay.io/prometheus/node-exporter:v1.5.0|
-| Metrics | Prometheus | quay.io/prometheus/prometheus:v2.44.0|
-| Metrics | Prometheus Operator | quay.io/prometheus-operator/prometheus-operator:v0.65.1|
-| Metrics | Prometheus Operator (config reloader) | quay.io/prometheus-operator/prometheus-config-reloader:v0.65.1|
-| Metrics | Prometheus Pushgateway | docker.io/prom/pushgateway:v1.5.1|
-| Metrics | Sidecar | quay.io/kiwigrid/k8s-sidecar:1.24.0|
+| Metrics | Kube State Exporter | k8s.gcr.io/kube-state-metrics/kube-state-metrics:v2.9.2|
+| Metrics | Node Exporter | quay.io/prometheus/node-exporter:v1.6.0|
```
Comment (gsmith-sas) [inline ARTIFACT_INVENTORY.md:31]: Incorrect version.

File: ARTIFACT_INVENTORY.md [hunk 14]
```diff
@@ -25,15 +25,15 @@ registry/repository/image_name:version
...
+| Metrics | Prometheus | quay.io/prometheus/prometheus:v2.45.0|
+| Metrics | Prometheus Operator | quay.io/prometheus-operator/prometheus-operator:v0.66.0|
+| Metrics | Prometheus Operator (config reloader) | quay.io/prometheus-operator/prometheus-config-reloader:v0.66.0|
+| Metrics | Prometheus Pushgateway | docker.io/prom/pushgateway:v1.6.0|
+| Metrics | Sidecar | quay.io/kiwigrid/k8s-sidecar:1.24.6|
```
Comment (gsmith-sas) [inline ARTIFACT_INVENTORY.md:36]: - Incorrect version.
- Please change "Sidecar" to "Auto-load Sidecars (Grafana)"

File: ARTIFACT_INVENTORY.md [hunk 15]
```diff
@@ -25,15 +25,15 @@ registry/repository/image_name:version
 | Metrics | Alertmanager | quay.io/prometheus/alertmanager:v0.25.0|
 | Metrics | BusyBox | docker.io/library/busybox:1.31.1|
 | Metrics | Ghostunnel | docker.io/ghostunnel/ghostunnel:v1.7.1|
-| Metrics | Grafana | docker.io/grafana/grafana:9.5.5|
+| Metrics | Grafana | docker.io/grafana/grafana:10.0.3|
 | Metrics | Ingress-NGINX Controller | registry.k8s.io/ingress-nginx/kube-webhook-certgen:v20221220-controller-v1.5.1-58-g787ea74b6|
```
Comment (gsmith-sas) [inline ARTIFACT_INVENTORY.md:29]: Please change "Ingress-NGINX Controller" to "Admission Webhook".

File: ARTIFACT_INVENTORY.md [hunk 16]
```diff
@@ -25,15 +25,15 @@ registry/repository/image_name:version
 | Metrics | Alertmanager | quay.io/prometheus/alertmanager:v0.25.0|
 | Metrics | BusyBox | docker.io/library/busybox:1.31.1|
```
Comment (gsmith-sas) [inline ARTIFACT_INVENTORY.md:26]: Please change 
- "BusyBox" to "BusyBox (OpenSearch)" (for the reference in the Logging subsystem).
- "BusyBox" to "BusyBox (Grafana)" (for the reference in the Metrics subsystem).

**Review by gsmith-sas — CHANGES_REQUESTED** [review]
Looks like some version numbers need to be updated in the ARTIFACT_INVENTORY.md and CHANGELOG.md files to reflect actual current behavior.  Also, I've also identified some tweaks to the wording used to identify some of the components in the ARTIFACT_INVENTORY.md file since you will updating that file anyway.

File: ARTIFACT_INVENTORY.md [hunk 17]
```diff
@@ -25,15 +25,15 @@ registry/repository/image_name:version
...
+| Metrics | Grafana | docker.io/grafana/grafana:10.0.3|
 | Metrics | Ingress-NGINX Controller | registry.k8s.io/ingress-nginx/kube-webhook-certgen:v20221220-controller-v1.5.1-58-g787ea74b6|
-| Metrics | Kube State Exporter | k8s.gcr.io/kube-state-metrics/kube-state-metrics:v2.8.2|
-| Metrics | Node Exporter | quay.io/prometheus/node-exporter:v1.5.0|
-| Metrics | Prometheus | quay.io/prometheus/prometheus:v2.44.0|
-| Metrics | Prometheus Operator | quay.io/prometheus-operator/prometheus-operator:v0.65.1|
-| Metrics | Prometheus Operator (config reloader) | quay.io/prometheus-operator/prometheus-config-reloader:v0.65.1|
-| Metrics | Prometheus Pushgateway | docker.io/prom/pushgateway:v1.5.1|
-| Metrics | Sidecar | quay.io/kiwigrid/k8s-sidecar:1.24.0|
+| Metrics | Kube State Exporter | k8s.gcr.io/kube-state-metrics/kube-state-metrics:v2.9.2|
```
[the remark this answers is not in the document]
Comment (gsmith-sas) [reply ARTIFACT_INVENTORY.md:30]: Still needed

File: ARTIFACT_INVENTORY.md [hunk 18]
```diff
@@ -25,15 +25,15 @@ registry/repository/image_name:version
...
+| Metrics | Node Exporter | quay.io/prometheus/node-exporter:v1.6.0|
+| Metrics | Prometheus | quay.io/prometheus/prometheus:v2.45.0|
+| Metrics | Prometheus Operator | quay.io/prometheus-operator/prometheus-operator:v0.66.0|
```
[the remark this answers is not in the document]
Comment (gsmith-sas) [reply ARTIFACT_INVENTORY.md:33]: still needed...should be v0.67.0, shouldn't it?

File: monitoring/values-prom-operator.yaml (hunk 10 above)
Comment (gsmith-sas) [reply monitoring/values-prom-operator.yaml:23]: CHANGELOG.md and component_version.env files still show v0.66.0.

**Review by gsmith-sas — CHANGES_REQUESTED** [review]
Sorry, still some incorrect version references in the the markdown and .env files compared to the yaml files.  I'm assuming the yaml files are correct.

**Review by gsmith-sas — APPROVED** [review]
I think we're good.  Approved.

**Pull request merged**